### PR TITLE
apparmor: add tests for generated template, add "header comment", and cleanup whitespace in generated profile

### PR DESCRIPTION
**Pull request by thaJeztah**
- related: https://github.com/containerd/containerd/pull/13270

### CODE REVIEW

File: apparmor/testdata/with-abstractions-base.golden [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  file,
+  umount,
+  # Host (privileged) processes may send signals to container processes.
+  signal (receive) peer=unconfined,
+  # runc may send signals to container processes (for "docker stop").
+  signal (receive) peer=runc,
+  # crun may send signals to container processes (for "docker stop" when used with crun OCI runtime).
+  signal (receive) peer=crun,
+  # dockerd may send signals to container processes (for "docker kill").
+  signal (receive) peer=,
```
Comment (thaJeztah) [inline apparmor/testdata/with-abstractions-base.golden]: We should probably make this either default to "unconfined", or make this rule conditional (skip if not set)

File: apparmor/testdata/with-abstractions-base.golden [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  # Container processes may send signals amongst themselves.
+  signal (send,receive) peer=abstractions-base,
```
Comment (thaJeztah) [inline apparmor/testdata/with-abstractions-base.golden:22]: Same for this one; possibly conditional or produce an error if empty.

**Review by vvoland — APPROVED** [review]
